library/platform_util.c

```diff
@@ -165,13 +165,37 @@ uint32_t mbedtls_platform_random_in_range( size_t num )
 #endif
 }
 
-/* Some compilers (armcc 5 for example) optimize away successive reads from a
- * volatile local variable (which we use as a counter-measure to fault
- * injection attacks), unless there is a call to an external function between
- * them. This functions doesn't need to do anything, it just needs to be
- * in another compilation unit. So here's a function that does nothing. */
-void mbedtls_platform_enforce_volatile_reads( void )
+int mbedtls_platform_random_delay( size_t max_rand )
```

Comment:
Suggestion - I don't think we need to parameterise this function. Normally, adding a parameter to a function makes better more flexible code, but here, it creates a new angle for potential vulnerabilities.

In real world usage, there's no reason we couldn't make this a compile time option, and fix what the maximum delay is. I don't think there's any reason why we'd vary it per call, nor at runtime.

Therefore I suggest you remove `max_rand` and replace it with a constant `MBEDTLS_MAX_RAND_DELAY`.

Reply:
mbedtls_platform_random_delay signature changed in the next commit, this also saves 44 bytes of ROM.